### ISSUE TITLE
ssh connection problems at some point in creation of cluster -  version 1.3.dev0 

### ISSUE THREAD

**Issue by maikenp** [gc3-uzh-ch/elasticluster#513]
As of last week, when creating a new cluster, the scripts run a while, but then at some point the ssh connection does not work and the cluster fails. 
The full log is attached as well as the config file and my keystone file which I source. 
[keystone.txt](https://github.com/gc3-uzh-ch/elasticluster/files/1611577/keystone.txt)
[elasticluster_config.txt](https://github.com/gc3-uzh-ch/elasticluster/files/1611578/elasticluster_config.txt)
[ssh_new.txt](https://github.com/gc3-uzh-ch/elasticluster/files/1611579/ssh_new.txt)

Using elasticluster version 1.3.dev0 

```(elasticluster)[centos@elasticluster-final ~]$ elasticluster -vvv ssh gridcluster-local
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Checking section `cluster/slurm` ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Checking section `login/centos` ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Checking section `setup/ansible-slurm` ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Checking section `cloud/iaas` ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Using class <class 'elasticluster.providers.ansible_provider.AnsibleSetupProvider'> from module <module 'elasticluster.providers.ansible_provider' from '/home/centos/elasticluster/src/elasticluster/providers/ansible_provider.pyc'> to instanciate provider 'ansible'
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG setting variable multiuser_cluster=no for node kind frontend
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG setting variable multiuser_cluster=no for node kind compute
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] WARNING ImportWarning: Not importing directory '/home/centos/elasticluster/lib/python2.7/site-packages/oslo_i18n/locale': missing __init__.py
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Using class <class 'elasticluster.providers.openstack.OpenStackCloudProvider'> from module <module 'elasticluster.providers.openstack' from '/home/centos/elasticluster/src/elasticluster/providers/openstack.pyc'> to instanciate provider 'openstack'
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack auth URL taken from env variable OS_AUTH_URL
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack user name taken from env variable OS_USERNAME
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack user domain name taken from env variable OS_USER_DOMAIN_NAME
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack password taken from env variable OS_PASSWORD
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack project name taken from env variable OS_PROJECT_NAME
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack project domain name taken from env variable OS_PROJECT_DOMAIN_NAME
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG OpenStack region_name taken from env variable OS_REGION_NAME
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Initializing OpenStack API clients: OS_AUTH_URL='https://api.uh-iaas.no:5000/v3' OS_USERNAME='maiken.pedersen@usit.uio.no' OS_USER_DOMAIN_NAME='dataporten' OS_PROJECT_NAME='uio-test-hpc-grid' OS_PROJECT_DOMAIN_NAME='dataporten' OS_REGION_NAME='osl' OS_CACERT='None'
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] INFO Using Keystone API v3 session to authenticate to OpenStack
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Creating OpenStack Compute API (Nova) v2 client ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Creating OpenStack Network API (Neutron) client ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Creating OpenStack Image API (Glance) v2 client ...
2018-01-08 13:04:15 elasticluster-final gc3.elasticluster[3898] DEBUG Creating OpenStack Volume API (Cinder) v2 client ...
2018-01-08 13:04:23 elasticluster-final gc3.elasticluster[3898] DEBUG Running command `ssh -i /home/centos/.ssh/cloud.key -o UserKnownHostsFile=/home/centos/.elasticluster/storage/gridcluster-local.known_hosts -o StrictHostKeyChecking=yes -p 22 centos@158.39.75.198`
Read from socket failed: Connection reset by peer
```

elasticluster -v start slurm -n gridcluster-local

Extract from where the creation of the cluster fails, otherwise for the rest of the log see the attachements. 
 ```
TASK [slurm-master : Add default user to cluster] **************************************************************************************************************************************************
changed: [frontend001] => (item=centos) => {"changed": true, "cmd": "sacctmgr --immediate --parsable --noheader list user Account=root | grep '^centos|' || sacctmgr --immediate --quiet add user 'centos' DefaultAccou\
nt=root", "delta": "0:00:00.748609", "end": "2018-01-08 11:37:32.538035", "item": "centos", "rc": 0, "start": "2018-01-08 11:37:31.789426", "stderr": "", "stderr_lines": [], "stdout": " Adding User(s)\n  centos\n Se\
ttings =\n  Default Account = root\n Associations =\n  U = centos    A = root       C = elasticlus", "stdout_lines": [" Adding User(s)", "  centos", " Settings =", "  Default Account = root", " Associations =", "  U\
 = centos    A = root       C = elasticlus"]}

TASK [slurm-master : Ensure `slurmctld` is running] ************************************************************************************************************************************************
changed: [frontend001] => (item=slurm) => {"changed": true, "enabled": true, "item": "slurm", "name": "slurm", "state": "started"}

RUNNING HANDLER [slurm-common : restart munge] *****************************************************************************************************************************************************
changed: [frontend001] => {"changed": true, "name": "munge", "state": "started"}

RUNNING HANDLER [slurm-master : restart slurmdbd] **************************************************************************************************************************************************
changed: [frontend001] => {"changed": true, "name": "slurmdbd", "state": "started"}

PLAY [Slurm worker nodes Playbook] *****************************************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************************************************
fatal: [compute001]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to remote host \"158.39.75.166\". Make sure this host can be reached over ssh", "unreachable": true}
       to retry, use: --limit @/home/centos/elasticluster/src/elasticluster/share/playbooks/site.retry

PLAY RECAP *****************************************************************************************************************************************************************************************
compute001                 : ok=37   changed=5    unreachable=1    failed=0
frontend001                : ok=104  changed=59   unreachable=0    failed=0

```

**Comment by riccardomurri** [comment]
I think I have found the problem -- testing now.

**Comment by riccardomurri** [comment]
Should be fixed in the latest "master" (commit 6a59058) -- can you please check?

**Comment by maikenp** [comment]
Thanks I will!

**Comment by maikenp** [comment]
Yes, that seems to have made it beyond the earlier point of failure. However, now I have problems with  the fstab part. Not sure if this is something related at all, or just something wrong on my side. I am investigating now. 

```
TASK [nfs-client : Ensure `portmap` is running (Ubuntu prior to 14.04)] ***************************************************************************************************************************************************************************
skipping: [compute001] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}

TASK [nfs-client : Ensure `rpcbind` is running (Ubuntu 14.04 or newer)] ***************************************************************************************************************************************************************************
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: ({{is_ubuntu}} and {{ansible_distribution_major_version}}|int >= 14)

skipping: [compute001] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}

TASK [nfs-client : Mount NFS filesystems] *********************************************************************************************************************************************************************************************************
included: /home/centos/elasticluster/src/elasticluster/share/playbooks/roles/nfs-client/tasks/nfsmount.yml for compute001

TASK [nfs-client : ensure /home directory exists] *************************************************************************************************************************************************************************************************
ok: [compute001] => {"changed": false, "gid": 0, "group": "root", "mode": "0755", "owner": "root", "path": "/home", "secontext": "system_u:object_r:home_root_t:s0", "size": 4096, "state": "directory", "uid": 0}

TASK [nfs-client : add to /etc/fstab] *************************************************************************************************************************************************************************************************************
fatal: [compute001]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: Connection timed out\n"}
	to retry, use: --limit @/home/centos/elasticluster/src/elasticluster/share/playbooks/site.retry

```

**Comment by riccardomurri** [comment]
> However, now I have problems with the fstab part.

Well, that would be another issue anyway so I'm closing this one.

Fixed in commit 6a59058.
